### PR TITLE
Use current PoST settings as default for Benchmarks

### PR DESCRIPTION
**Pull request by fasmat**
PoST now uses 16 byte labels and N = 8192 for initialization. This updates the benchmark to be in line with those parameters.

### CODE REVIEW

**Review by pigmej — CHANGES_REQUESTED** [review]
Imho all tests in that file needs that adjustment. Most of them expose CLI so we need to adjust them all.

**Comment by fasmat** [comment]
> Imho all tests in that file needs that adjustment. Most of them expose CLI so we need to adjust them all.

All tests have been updated with N = 8192. LabelSize is a parameter that can be passed as an argument and now defaults to 128 (=16 bytes).

Which other adjustments do you think we need?

File: test/test.cpp [hunk 1]
```diff
@@ -424,7 +424,7 @@ int do_test_pow(uint64_t aStartPos, int aLabelsCount, unsigned aDiff, unsigned a
 
 		if (spacemesh_api_get_providers(providers, providersCount) == providersCount) {
```
Comment (fasmat) [inline test/test.cpp:425]: This test has testvectors that need updating before we can change N = 512

File: test/test.cpp [hunk 2]
```diff
@@ -652,7 +652,7 @@ bool do_test_vector(const TestVector *aTestVector, bool aPrintResult)
 					uint64_t hashes_computed;
 					uint64_t hashes_per_sec;
 					uint64_t idx_solution = -1;
-					uint8_t D[32] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+					uint8_t D[32] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
 
 					std::unique_ptr<uint8_t> out((uint8_t*)calloc(1, labelsBufferSize));
```
Comment (fasmat) [inline test/test.cpp:657]: This function also needs to be updated with N = 8192.

File: test/test.cpp [hunk 3]
```diff
@@ -562,14 +562,14 @@ int do_test_leafs_and_pow(uint64_t aStartPos, int labelSize, int aLabelsCount, u
 					uint64_t hashes_computed;
 					uint64_t hashes_per_sec;
 					printf("%s: ", providers[i].model);
-					int status = scryptPositions(providers[i].id, s_id, aStartPos, aStartPos + aLabelsCount - 1, 8, s_salt, SPACEMESH_API_COMPUTE_LEAFS | SPACEMESH_API_COMPUTE_POW, out, 8192, 1, 1, D, &idx_solution, &hashes_computed, &hashes_per_sec);
+					int status = scryptPositions(providers[i].id, s_id, aStartPos, aStartPos + aLabelsCount - 1, 8, s_salt, SPACEMESH_API_COMPUTE_LEAFS | SPACEMESH_API_COMPUTE_POW, out, 512, 1, 1, D, &idx_solution, &hashes_computed, &hashes_per_sec);
```
Comment (fasmat) [inline test/test.cpp]: Still 512 because of golden testvector

**Comment by pigmej** [comment]
@fasmat I think we might need to add some info to the README.md about that change next to the benchmarks so people are not confused (having 16 less)

**Comment by fasmat** [comment]
Benchmarks are now tracked with MB/s instead of MH/s, since data throughput is a better measure to capture for a benchmark.

I also updated the README in 2 ways: newer benchmarks are printed higher up in the table and results are tracked with MB/s

**Review by pigmej — APPROVED** [review]

**Comment by pigmej** [comment]
`Apple M2 Pro: status 0, 100000 labels, 37193 labels/s, 0.57 MiB/s` I like that output :)


but @fasmat we need to fix unit tests ;)

**Comment by fasmat** [comment]
there seems to be an issue with the arm64 build. I cannot get the one failing test to work on it 🙁

**Pull request closed without merging**